solver/src/settlement_submission.rs

```diff
@@ -31,13 +33,22 @@ pub async fn submit(
     gas_price_cap: f64,
     settlement: Settlement,
 ) -> Result<()> {
-    let nonce = transaction_count(contract)
-        .await
-        .context("failed to get transaction_count")?;
     let settlement = encode_settlement(&settlement)?;
     // Check that a simulation of the transaction works before submitting it.
     simulate_settlement(&settlement, contract).await?;
 
+    let nonce = transaction_count(contract)
+        .await
+        .context("failed to get transaction_count")?;
+    let address = match &contract.defaults().from {
+        Some(account) => account.address(),
+        None => return Err(anyhow!("no default sender address")),
```

Comment:
Curious on why error here instead of crash? Wouldn't it be a non-recoverable configuration error if from is not set?

Reply:
I wasn't sure what was more appropriate. Changed it to panic now.